src/frontend/src/handler/drop_table.rs

```diff
@@ -57,10 +61,75 @@ pub async fn handle_drop_table(
         if table.table_type() != TableType::Table {
             return Err(table.bad_drop_error());
         }
-
-        (table.associated_source_id(), table.id())
+        (table.associated_source_id(), table.id(), table.engine)
     };
 
+    match engine {
+        Engine::Iceberg => {
+            let source = session
+                .env()
+                .catalog_reader()
+                .read_guard()
+                .get_source_by_name(
+                    db_name,
+                    schema_path,
+                    &(ICEBERG_SOURCE_PREFIX.to_string() + &table_name),
+                )
+                .map(|(source, _)| source.clone())?;
+
+            // Drop sink
+            // Drop iceberg table
+            //   - Purge table from warehouse
+            //   - Drop table from catalog
+            // Drop source
+            crate::handler::drop_sink::handle_drop_sink(
+                handler_args.clone(),
+                ObjectName::from(vec![Ident::from(
+                    (ICEBERG_SINK_PREFIX.to_string() + &table_name).as_str(),
+                )]),
+                true,
+                false,
+            )
+            .await?;
+
+            let config = ConnectorProperties::extract(source.with_properties.clone(), false)?;
+            if let ConnectorProperties::Iceberg(iceberg_properties) = config {
```

Comment:
Add some todo comments:
1. Repeatable deletion to ensure system robustness in case of deletion failure
2. Consider the order of deletion, the source should be deleted first to ensure that other sessions are no longer readable.